### PR TITLE
fix: Bug fixed for add filter button in production

### LINKED ISSUE #4254: [NV-2350] Should show filter modal in production
## Reproduction steps
In production the filters button is disabled. A user can not view the filters applied for him. 

1. Promote the workflow with any steps that have filters applied to the production environment.
2. Change the environment to production.
3. Navigate to the promoted workflow and open the step that has filters applied.
4. The filters button is disabled

image

## Definition of Done
- The filters button should be enabled but when clicked all the Filter Modal fields should be disabled

### PR DESCRIPTION
**Pull request by sjcode99**
### What change does this PR introduce?
This PR fixes the bug for the Add Filter button in production

The change was to enable the add filter button in the production environment.

Fixes #4254

### CODE REVIEW

**Review by LetItRock — CHANGES_REQUESTED** [review]
@sjcode99 left you a comment in the original ticket ;)

**Comment by sjcode99** [comment]
@LetItRock  updated the PR with the latest changes. Please verify it . Thanks

**Review by LetItRock — DISMISSED** [review]
hey @sjcode99 👋 
almost there, you are doing great! 🙌 just noticed that we didn't cover all the filter cases, please take a look at the below screenshot ;)
Screenshot 2023-10-02 at 17 45 53

**Comment by sjcode99** [comment]
@LetItRock, Ohh!!😮 I have missed this. Changes update, Please have a look.

File: apps/web/src/pages/templates/filter/FilterModal.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { DeleteStepButton, FilterButton } from './FilterModal.styles';
 import { OnlineFiltersForms } from './OnlineFiltersForms';
 import { PreviousStepFiltersForm } from './PreviousStepFiltersForm';
 import { useMemo } from 'react';
+import { boolean } from 'zod';
```
Comment (scopsy) [inline apps/web/src/pages/templates/filter/FilterModal.tsx:12]: This seems like it was imported by a mistake.

File: apps/web/src/pages/templates/workflow/SideBar/StepSettings.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ export function StepSettings({ index }: { index: number }) {
           onClick={() => {
             setFilterOpen(true);
           }}
-          disabled={readonly}
+          disabled={false}
```
Comment (scopsy) [inline apps/web/src/pages/templates/workflow/SideBar/StepSettings.tsx:47]:
```suggestion
```

**Review by scopsy — CHANGES_REQUESTED** [review]
Thank you @sjcode99 looking good! Just a small thing here to fix this un-needed import

**Comment by sjcode99** [comment]
Hey @scopsy, removed the unnecessary import. Please have a look

**Review by scopsy — APPROVED** [review]

**Comment by sjcode99** [comment]
Hey @LetItRock  @scopsy . Seems my code is not yet merged. can you please help with this merge

**Comment by sjcode99** [comment]
Hey @LetItRock, it's been a long since I created PR and I can still see it's not merged. Is there any issue in the code which needs to be addressed?
Thanks :)

**Pull request merged**